src/dataflow/src/sink/kafka.rs

```diff
@@ -143,9 +143,30 @@ struct KafkaSink {
     encoder: Encoder,
     producer: ThreadedProducer<SinkProducerContext>,
     activator: timely::scheduling::Activator,
+    txn_timeout: Duration,
 }
 
 impl KafkaSink {
+    fn transition_on_error(&self, current_state: SendState, e: KafkaError) -> SendState {
```

Comment:
might be worth adding a comment indicating that this has to be called on transactional errors (if that is indeed the case)?

Reply:
Good call! I was originally seeing if I could have this handle the error conditions in send() as well, and forgot to make the name less generic when I didn't do that